hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodiePositionBasedFileGroupRecordBuffer.java

```diff
@@ -123,6 +124,13 @@ public void processNextDataRecord(T record, Map<String, Object> metadata, Object
   @Override
   public void processDeleteBlock(HoodieDeleteBlock deleteBlock) throws IOException {
     List<Long> recordPositions = extractRecordPositions(deleteBlock);
+    if (recordMerger.getMergingStrategy().equals(DEFAULT_MERGER_STRATEGY_UUID)) {
+      for (Long recordPosition : recordPositions) {
+        records.put(recordPosition,
+            Pair.of(Option.empty(), readerContext.generateMetadataForRecord(null, "", 0)));
```

Comment:
potential NPE? let's add some test for this case.

Reply:
Added tests.